cli/command/stack/list_test.go

```diff
@@ -46,9 +47,11 @@ func TestListErrors(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		cmd := newListCommand(test.NewFakeCli(&fakeClient{
+		cli := test.NewFakeCli(&fakeClient{
 			serviceListFunc: tc.serviceListFunc,
-		}))
+		})
+		cli.SetClientInfo(func() command.ClientInfo { return command.ClientInfo{Orchestrator: "swarm"} })
```

Comment:
Should probably go in the `NewFakeCli` builder

```go
cmd := newListCommand(test.NewFakeCli(&fakeClient{
  // …
}, test.OrchestratorSwarm))
```

and defining in `internal/test` : 

```go
func OrchestratorSwarm(c *FakeCli) {
  cli.SetClientInfo(func() command.ClientInfo { return command.ClientInfo{Orchestrator: "swarm"} })
}

func OrchestratorKubernetes(c *FakeCli) {
  cli.SetClientInfo(func() command.ClientInfo { return command.ClientInfo{Orchestrator: "swarm"} })
}

func OrchestratorAll(c *FakeCli) {
  cli.SetClientInfo(func() command.ClientInfo { return command.ClientInfo{Orchestrator: "swarm"} })
}
```

nit: `SetClientInfo` should be removed (with the `clientInfoFunc`) and make a builder of `ClientInfo` — to have sthg like `test.WithClientInfo(OrchestratorSwarm)` or sthg)

Reply:
OK